services/iplant-groups/src/iplant_groups/clients/grouper.clj

```diff
@@ -520,35 +520,58 @@
 
 ;; Permission assignment
 ;; search/lookup
+(defn- format-attribute-def-lookup
+  [{:keys [attribute_def_id attribute_def]}]
+  (cond attribute_def_id [{:uuid attribute_def_id}]
+        attribute_def    [{:name attribute_def}]))
+
+(defn- format-attribute-def-name-lookup
+  [{:keys [attribute_def_name_id attribute_def_name]}]
+  (cond attribute_def_name_id [{:uuid attribute_def_name_id}]
+        attribute_def_name    [{:name attribute_def_name}]))
+
+(defn- format-role-lookup
+  [{:keys [role_id role]}]
+  (cond role_id [{:uuid role_id}]
+        role    [{:groupName role}]))
+
+(defn- format-action-names-lookup
+  [{:keys [action_names]}]
+  (when (seq action_names) action_names))
+
```

Comment:
These changes all enable the permission assignment lookup to use either names or IDs.